storage/engine/rocksdb/db.cc

```diff
@@ -1179,10 +1160,14 @@ class BaseDeltaIterator : public rocksdb::Iterator {
     return merged_.data == NULL;
```

Comment:
What's going on in l. 1154 to here? Why have we advanced past the last entry for key?